cluster-autoscaler/core/scale_down.go

```diff
@@ -434,7 +434,8 @@ func (sd *ScaleDown) TryToScaleDown(allNodes []*apiv1.Node, pods []*apiv1.Pod, p
 
 			nodeCPU, nodeMemory, err := getNodeCoresAndMemory(node)
 			if err != nil {
-				glog.Warningf("Error getting node resources: %v", err)
+				glog.Errorf("Error getting node resources: %v", err)
+				continue
```

Comment:
I'm not at all sure if we should exit here. In the unlikely case there's a node object with corrupted resource data, we may actually prefer to remove it (assuming its pods can be moved and group is above min size), rather than keep it in the cluster forever. While Cluster Autoscaler isn't responsible for healing the cluster, it already does some things like this (e.g. remove unregistered nodes.)

Reply:
I am not convinced. I agree with the fact that our responsibility is not healing cluster from nodes with broken definitions. 
Also it seems wrong to me to assume that resources are equal to 0 on some random error (which could be due to bug in our code for example).

Reply:
Looking at what scheduler does: https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/cache/node_info.go#L150
it seems similar (assuming 0 if there's no CPU in resource list.) So nothing will get scheduled on such node anyway (since minimal CPU requests for all containers are always set by scheduler, even if the user doesn't specify them.)

I think we can agree that deleting a corrupted node where nothing can be scheduled is good (regardless of whose responsibility it is.)

As for the bug in our code scenario, unfortunately it'll still be true that if there's a bug in how we calculate resources, we risk going outside the specified limits (e.g. if we return incorrect value and no error.)

Reply:
I could somewhat agree (but still have very mixed feelings about it) if we were sure that error here means broken nodes  (e.g. nodes which are lacking resource definitions). But we are not sure if that is the case. 
E.g. when we compute number of gpu's used by node we call `nodeGroup.TemplateNodeInfo()`. Which is some arbitrary interface function, which could for example do an API call. And can temporarily return error. I strongly feel that we should not perform any scale_down action for considered node in such case.

IMHO we should count the errors here. Have metric with it so we can spot that something bad is happening. And if such thing happens react on that. Alternative to explicit metric could be a graph based on log grepping.